src/QsCompiler/QsTargeting/Target.fs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Targeting
+
+open Microsoft.Quantum.QsCompiler.SymbolManagement
+open System.Collections.Generic
+open System.Threading
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Diagnostics
+
+type Target(builtIns : NamespaceManager, capabilityLevel : CapabilityLevel) =
+    member this.BuiltIns with get () = builtIns
+
+    member this.CapabilityLevel with get () = capabilityLevel
+
+    member this.ValidateSpecialization (spec : QsSpecialization) : List<DiagnosticItem> = 
+        let diagnostics = new List<DiagnosticItem>()
+        if spec.RequiredCapability > this.CapabilityLevel
+        then diagnostics.Add(Error ErrorCode.UnexpectedCompilerException)
```

Comment:
Are these capability levels something that is documented for the public to see? If so, a more specific error message here could be useful.

Reply:
I'm going to remove this file from the PR, and add it later.
Yes, they will be public.